genstudio-external-template-app/src/genstudiopem/web-src/src/components/TemplateCard.tsx

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2025 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import React, { useState } from "react";
+import { View, Image, Text } from "@adobe/react-spectrum";
+import { DamAsset } from "../types";
+
+interface TemplateCardProps {
+    template: DamAsset;
+    isSelected: boolean;
+    onSelect: (asset: DamAsset) => void;
+}
+
+export default function TemplateCard({ template, isSelected, onSelect }: TemplateCardProps): JSX.Element {
+  const [isHovered, setIsHovered] = useState(false);
+
+  const cardStyles: React.CSSProperties = {
+    cursor: "pointer",
+    position: "relative",
+    width: "230px",
+    height: "241px",
+    borderRadius: "var(--spectrum-global-dimension-size-50)",
+    backgroundColor: "var(--spectrum-global-color-gray-50)",
+    overflow: "hidden",
+    transition: "box-shadow 0.2s ease-out",
+    transform: "translateZ(0)",
+    willChange: "box-shadow",
+    border: isSelected ? "2px solid var(--spectrum-global-color-gray-900)" : "2px solid transparent",
+    margin: "0px",
+    boxShadow: isSelected
```

Comment:
We should not rely on CSS property which can complicate things.
I would rather use something in react-spectrum instead to indicate selection.
Note that usually just showing the check mark in the toggle is enough.